src/parse.rs

```diff
@@ -682,20 +684,24 @@ fn process_element<'input>(
             let prefix = prefix.as_str();
             let local = local.as_str();
 
-            doc.nodes[parent_id.get_usize()].range.end = token_span.end() as u32;
-            if let NodeKind::Element { ref tag_name, .. } = doc.nodes[parent_id.get_usize()].kind {
-                if prefix != tag_name.prefix || local != tag_name.name {
+            let parent_node = &mut doc.nodes[parent_id.get_usize()];
+            let parent_prefix = *parent_prefixes.last().unwrap();
```

Comment:
Do we guarantee that prefixes are not empty?

Reply:
I changed the first commit to check

```rust
debug_assert!(!parent_prefixes.is_empty());
```

instead of

```rust
debug_assert!(parent_prefixes.pop().unwrap());
```

which never fired during the test and I do initialize it with `vec![""]` (which would be the prefix of the root node). But I have admittedly no proof besides always pushing one prefix when I update the parent ID as to why this `unwrap` is unreachable.

I could do `unwrap_or("")` instead but I feared that this would hide mistakes on my part?

Reply:
Oh, I see now. `parent_prefixes` is never empty. Then `unwrap` is fine. Maybe a comment before `unwrap` would be nice.

Reply:
> Maybe a comment before unwrap would be nice.

Done